### ISSUE TITLE
Unable to perform chinese language OCR using ocrmypdf-polyglot

### ISSUE THREAD

**Issue by wallclock** [jbarlow83/OCRmyPDF#81]
Using the lastest ocrmypdf-polyglot docker image, I ran `docker run -v "$(pwd):/home/docker" jbarlow83/ocrmypdf-polyglot -v -l chi_sim textbook.pdf textbook-ocr.pdf` according to instructions on a single page A4 simplified chinese PDF with ~1.5 line spacing. The result was mainly solid black square characters (`■■■■■`).

Is there a reference PDF for which I can test on for simplified chinese (or any asian language)?

**Comment by jbarlow83** [comment]
Could you send me one representation page from the file you are trying to OCR?

**Comment by wallclock** [comment]
Here you go: [textbook2.pdf](https://github.com/jbarlow83/OCRmyPDF/files/385047/textbook2.pdf)

**Comment by jbarlow83** [comment]
Thanks for this. The tesseract OCR engine is working more or less correctly.

This problem is caused by an Unicode handling issue in Ghostscript, which ocrmypdf uses. Ghostscript has fixed the bug but the fix is only available in the development version. Because this same bug has caused some other recent issues I am looking into a workaround. For the moment, it does not work Simplified Chinese and will probably have trouble with most other Asian languages.

**Comment by jbarlow83** [comment]
Note to self: this issue is probably blocked by https://github.com/mstamy2/PyPDF2/issues/252

**Comment by jbarlow83** [comment]
Note to self: hocr renderer also botches Chinese text because the font selected (Helvetica) only does not include Chinese characters. See reportlab manual for how to set up Asian fonts. Tesseract renderer does fine.

**Comment by wallclock** [comment]
Thanks for the explanation! Ghostscript seems to have a 6-month release cycle; shall keep my fingers crossed.

**Comment by jbarlow83** [comment]
存轻乡圣王里人的领导力

v4.2 offers a partial solution if you specify the tesseract renderer and regular PDF output as opposed to PDF/A. This bypasses two of the bugs described above.
```
ocrmypdf -f -l chi_sim --pdf-renderer tesseract --output-type pdf   input.pdf output.pdf
```

You might need to rescan that image if the OCR was not what you hoped for. There is some warping in the text baseline. (I can't help with OCR quality, that's an issue for tesseract-ocr).

**Comment by wallclock** [comment]
Great! I will be happy to try out the new release. Unfortunately, the latest polyglot docker image released 2 days ago seems to be version 4.1.3.

**Comment by jbarlow83** [comment]
I tried manually retriggering the Docker build. (It's usually automatic.)
Hopefully that will do it.
